Core/include/Acts/TrackFinder/CombinatorialKalmanFilter.hpp

```diff
@@ -0,0 +1,1255 @@
+            auto trackTip =
+                result.fittedStates.addTrackState(stateMask, result.currentTip);
+
+            // Get the track state proxy
+            auto trackStateProxy = result.fittedStates.getTrackState(trackTip);
+
+            // Fill the track state proxy
+            if (predictedShared) {
+              // The predicted parameter is already stored, just set the index
+              auto sharedPredicted =
+                  result.fittedStates.getTrackState(neighborTip);
+              trackStateProxy.data().ipredicted =
+                  sharedPredicted.data().ipredicted;
+            } else {
+              trackStateProxy.predicted() = boundParams.parameters();
+              trackStateProxy.predictedCovariance() = *boundParams.covariance();
+            }
+            trackStateProxy.jacobian() = jacobian;
+            trackStateProxy.pathLength() = pathLength;
+
+            // Get and set the type flags
+            auto& typeFlags = trackStateProxy.typeFlags();
+            typeFlags.set(TrackStateFlag::MaterialFlag);
+            typeFlags.set(TrackStateFlag::ParameterFlag);
+            if (isOutlier) {
+              // Set the outlier type flag
+              typeFlags.set(TrackStateFlag::OutlierFlag);
+            } else {
+              // Set the measurement type flag
+              typeFlags.set(TrackStateFlag::MeasurementFlag);
+            }
+
+            // Assign the source link and calibrated measurement to the track
+            // state
+            if (sourcelinkShared) {
+              // The source link is already stored, just set the index
+              auto sharedMeasurement =
+                  result.fittedStates.getTrackState(index_it->second);
+              trackStateProxy.data().iuncalibrated =
+                  sharedMeasurement.data().iuncalibrated;
+            } else {
+              trackStateProxy.uncalibrated() = sourcelinks.at(index);
+            }
+            std::visit(
+                [&](const auto& calibrated) {
+                  trackStateProxy.setCalibrated(calibrated);
+                },
+                m_calibrator(trackStateProxy.uncalibrated(),
+                             trackStateProxy.predicted()));
+
+            // Set the filtered parameter
+            if (isOutlier) {
+              // No Kalman update for outlier
+              // Set the filtered parameter index to be the same with predicted
+              // parameter
+              trackStateProxy.data().ifiltered =
+                  trackStateProxy.data().ipredicted;
+
+              ACTS_VERBOSE(
+                  "Creating outlier track state with tip = " << trackTip);
+
+              // Count the number of processedStates, measurements, outliers and
+              // holes
+              auto tipState =
+                  TipState{preTipState.nStates + 1, preTipState.nMeasurements,
+                           preTipState.nOutliers + 1, preTipState.nHoles};
```

Comment:
Ok nitpick again, but using `auto` here seems weird:

```suggestion
              TipState tipState{preTipState.nStates + 1, preTipState.nMeasurements,
                                preTipState.nOutliers + 1, preTipState.nHoles};
```

Reply:
Thanks. Changed.